theano/sandbox/cuda/type.py

```diff
@@ -21,6 +21,11 @@
     import cuda_ndarray.cuda_ndarray as cuda
     from theano.sandbox.cuda.nvcc_compiler import NVCC_compiler
     import cuda_ndarray
+    # Python 3 does not necessarily set __file__. May need manual setting
+    try:
+        cuda_ndarray.__file__
+    except AttributeError:
+        cuda_ndarray.__file__ = os.path.join(cuda_ndarray.__path__._path[0], 'cuda_ndarray.pyd')
```

Comment:
Comment added.
I don't know whether the problem will occur on other platforms. You reminded me that using hardcoded `.pyd` extension is wrong, and I have changed it into `get_lib_extension()` which can determine the correct extension on different platforms.